cmake/sdlcompilers.cmake

```diff
@@ -160,3 +160,55 @@ function(SDL_AddCommonCompilerFlags TARGET)
     endif()
   endif()
 endfunction()
+
+function(check_x86_source_compiles FUNC_DECL MAIN_BODY VAR)
+  if(ARGN)
+    message(FATAL_ERROR "Unknown arguments: ${ARGN}")
+  endif()
+  check_c_source_compiles("
+    #ifdef __APPLE__
+    # include \"TargetConditionals.h\"
+    # if TARGET_CPU_X86 || TARGET_CPU_X86_64
+    #  define test_enabled 1
+    # else
+    #  define test_enabled 0
+    # endif
+    #else
+    # define test_enabled 1
+    #endif
+    #if test_enabled
+    ${FUNC_DECL}
+    #endif
+    int main(int argc, char *argv[]) {
+    # if test_enabled
+      ${MAIN_BODY}
+    # endif
+      return 0;
```

Comment:
If I'm reading this correctly, this x86 test will succeed for aarch64-apple (non-x86.) Similarly, the arm test below will succeed for x86_64-apple (non-arm.) Is this what you really intend?

Reply:
These functions will only get called if a fat library contains x64/x86.
I'll add a fail-safe that returns false when the cpu is not x86/arm.